smart-hadoop-support/smart-hadoop-client/src/main/java/org/smartdata/hdfs/client/SmartDFSClient.java

```diff
@@ -129,13 +133,18 @@ public DFSInputStream open(String src, int buffersize,
       boolean verifyChecksum)
       throws IOException, UnresolvedLinkException {
     FileState fileState;
+    src = checkS3ForXattr(src);
     if (healthy) { // TODO: required indeed
       fileState = smartClient.getFileState(src);
       if (fileState.getFileStage().equals(FileState.FileStage.PROCESSING)) {
         throw new IOException("Cannot open " + src + " when it is under PROCESSING to "
             + fileState.getFileType());
       }
-    } else {
+    } 
+    else if (src.startsWith("s3a")) {
```

Comment:
The same with previous comment. We can use FileType to identify file type rather than s3a prefix.